### ISSUE TITLE
BlitzCredoChecks.DocsBeforeSpecs does not take into account defp

### ISSUE THREAD

**Issue by jcowgar** [theblitzapp/blitz_credo_checks#77]
Dialyzer works fine and provides helpful information for spec's attached to a defp function. The DocsBeforeSpecs does not take a defp into account. So, it is resulting in some false hits. The scenario is a `@spec` attached to a `defp`, then later in the file a `@doc` `@spec` `def`. DocsBeforeSpecs sees the `@spec`, skips the `defp` it is attached to and then runs into the `@doc` which is actually actually attached to the next def line. It then throws a false warning.

```elixir
@spec hello(String.t()) :: :ok
defp hello(who) do
  IO.puts("Hello, #{who}")
end

@doc """
Do something
"""
@spec do_something(integer()) :: integer()
def do_something(val) do
  val
end
```

**Comment by alanvardy** [comment]
Good catch, thank you. I will address this next week.

**Comment by alanvardy** [comment]
Which version of Elixir are you running @jcowgar ? I'm having trouble reproduction this in the test suite and another codebase.

**Comment by jcowgar** [comment]
Good afternoon. I am using 1.15.7, Erlang 26.1.2. I created a simple mix new project that exhibits this problem: https://codeberg.org/jcowgar/hello-blitz-example

My output is: 
```
jeremy@guixd ~/Projects/elixir/hello$ mix credo
Checking 3 source files ...

  Code Readability                                                                                                                            
┃ 
┃ [R] ↗ @doc goes before @spec
┃       lib/hello.ex:20 #(Hello)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.04 seconds (0.03s to load, 0.01s running 56 checks on 3 files)
5 mods/funs, found 1 code readability issue.

Showing priority issues: ↑ ↗ →  (use `mix credo explain` to explain issues, `mix credo --help` for options).
jeremy@guixd ~/Projects/elixir/hello$ elixir  --version
Erlang/OTP 26 [erts-14.1.1] [source] [64-bit] [smp:32:32] [ds:32:32:10] [async-threads:1] [jit:ns]

Elixir 1.15.7 (compiled with Erlang/OTP 26)
```

**Comment by alanvardy** [comment]
I wanted to let you know that I haven't forgotten about you. Thank you for the example project. I plan to devote time to this after New Year's. It is currently crunch time at work, so I unfortunately have to triage my time.

**Comment by jcowgar** [comment]
Crunch time, what's that? :-D ... it's all good, thanks for the project!

**Comment by alanvardy** [comment]
Thanks for the steps to reproduce, v0.1.10 is published with the fix!
